src/claim/assign-name/assign-name.js

```diff
@@ -0,0 +1,65 @@
+const Claim = require('../claim');
+const utils = require('../../utils');
+
+/**
+ * @param  {Uint32} Version
+ * @param  {String} Hash name
+ * @param  {String} Address identifier
+ */
+class AssignName {
+  constructor(_version = 0, _hashName = '', _id = '') {
+    const versionBuff = Buffer.alloc(4);
+    versionBuff.writeUInt32BE(_version);
+    this.structure = {
+      claimType: utils.hashBytes('iden3.claim.assign_name').slice(24, 32),
+      version: versionBuff,
+      hashName: utils.hashBytes(_hashName).fill(0, 0, 1),
+      id: utils.hexToBytes(_id),
+    };
+  }
+
+  /**
+   * @returns {Object Elements} Element representation of the claim
+   */
+  elements() {
+    const element = new Claim.Elements();
+    let endIndex = element.e3.length;
```

Comment:
What does e3 mean? Could you provide some comments about what is the purpose of these lines of code?